src/main/java/io/neonbee/data/DataContext.java

```diff
@@ -96,6 +98,105 @@ public interface DataContext {
      */
     <T> T remove(String key);
 
+    /**
+     * Arbitrary response data of this context, which is passed through each processing step of the verticle backwards
+     * from callee to caller.
+     *
+     * @return all the context response data as a map.
+     */
+    Map<String, Object> responseData();
+
+    /**
+     * Sets all the response data.
+     *
+     * @param data the data, must be compatible to {@link JsonObject#JsonObject(Map)}
+     * @return a reference to this DataContext for chaining
+     */
+    DataContext setResponseData(Map<String, Object> data);
```

Comment:
not needed, everything that is to be changed on the responseData should be changed on the responseData() map itself, e.g. responseData().putAll(...)